packages/@react-spectrum/utils/src/BreakpointProvider.tsx

```diff
@@ -62,7 +62,15 @@ export function useMatchedBreakpoints(breakpoints: Breakpoints): string[] {
     }
 
     let onResize = () => {
-      setBreakpoint(getBreakpointHandler());
+      const breakpointHandler = getBreakpointHandler();
+
+      setBreakpoint(previousBreakpointHandler => {
+        if (previousBreakpointHandler.length !== breakpointHandler.length) {
```

Comment:
I was thinking of this as an option, however, the breakpoints themselves could change as well, which might make this comparison a false equivalence
We already loop over all the matched media, we can handle some comparison in there

I did not test this, but this was the line of my thinking for a change in getBreakpointHandler
```
let bqRef = useRef([]);
...
    let matched = [];
    for (let i in breakpointQueries) {
      let equivalent = true;
      let query = breakpointQueries[i];
      if (window.matchMedia(query).matches) {
        matched.push(entries[i][0]);
        if (bqRef.current.length < i && bqRef.current[i] !== entries[i][0]) {
          equivalent = false;
        }
      }
    }
    matched.push('base');
    if (!equivalent || bqRef.current.length !== matched.length) {
      bqRef.current = matched;
    }
    return bqRef.current;
```

Reply:
Yes, I understand that the predicate for the comparison is pretty simple, but it stands on the assumption that breakpoints wouldn't change. If that's not a solid assumption, then it we should approach this differently, as you've suggested. That said, I don't think the logic should live in `getBreakpointHandler`, though, because that's merely a helper function and doesn't necessarily need to know what the caller is planning to do with its result. Instead, I think the logic should live in the caller (i.e. `onResize`) since it's the one that knows what to do with the result and whether or not `setBreakpoint` should be called.

Reply:
Another thing is, I think that `if` statement should just say:
```ts
if (bqRef.current.length - 1 < i || bqRef.current[i] !== matched[i]) {
  break; // break immediately from the `for` loop
}
```
Later on, it should be enough to just check on the lengths since we've already checked the elements while building `matched`. If the lengths match, then we're good. Otherwise, we must've encountered a non-match.

But, I'm making a little adjustment in the place where we call `setBreakpoint`. Please have a look and let me know if you prefer to go the `ref` way.